qcfractal/interface/collections/dataset.py

```diff
@@ -521,64 +615,136 @@ def _default_parameters(self,
+        ----------
+        indexer : Dict[str, 'ObjectId']
+            A key/value index of molecules to query
+
+        Returns
+        -------
+        pd.Series
+            A series of Molecules
+
+        Raises
+        ------
+        KeyError
+            If no records match the query
+        """
+
+        molecules = []
+        molecule_ids = list(set(indexer.values()))
+        for i in range(0, len(molecule_ids), self.client.query_limit):
+            molecules.extend(self.client.query_molecules(id=molecule_ids[i:i + self.client.query_limit]))
+
+        molecules = pd.DataFrame.from_dict([{"molecule_id": x.id, "molecule": x} for x in molecules])
+        if len(molecules) == 0:
+            raise KeyError("Query matched 0 records.")
+
+        df = pd.DataFrame.from_dict(indexer, orient="index", columns=["molecule_id"])
+        df.reset_index(inplace=True)
+
+        # Outer join on left to merge duplicate molecules
+        df = df.merge(molecules, how="left", on="molecule_id")
+        df.set_index("index", inplace=True)
+        df.drop("molecule_id", axis=1, inplace=True)
+
+        return df
+
+    def _get_records(self,
+                     indexer: Dict[str, 'ObjectId'],
+                     query: Dict[str, Any],
+                     projection: Optional[Dict[str, bool]] = None,
+                     merge: bool = False,
+                     raise_on_plan: Union[str, bool] = False) -> 'pd.Series':
         """
         Runs a query based on an indexer which is index : molecule_id
 
         Parameters
         ----------
-        indexer : str
-            The primary index of the query
+        indexer : Dict[str, 'ObjectId']
+            A key/value index of molecules to query
         query : Dict[str, Any]
             A results query
-        field : str, optional
-            The field to pull from the ResultEntry
-        scale : str, optional
-            The scale of the computation
+        projection : Optional[Dict[str, bool]], optional
+            Description
+        merge : bool, optional
+            Sum compound queries together, useful for mixing results
+        raise_on_plan : Union[str, bool], optional
+            Raises a KeyError is True or string if a multi-stage plan is detected.
 
         Returns
         -------
-        Series
+        pd.Series
             A Series of the data results
+
         """
+        self._check_client()
         self._check_state()
 
-        field = field.lower()
-
         ret = []
-        for query_set in composition_planner(**query):
+        plan = composition_planner(**query)
+        if raise_on_plan and (len(plan) > 1):
+            if raise_on_plan is True:
+                raise KeyError("Recieved a multi-stage plan when this function does not support multi-staged plans.")
+            else:
+                raise KeyError(raise_on_plan)
+
+        for query_set in plan:
+
+            if isinstance(query_set["keywords"], str):
```

Comment:
This doesn't work when `query_set["keywords"]` is an id, which gets passed as a string. Example:
```
ds = client.get_collection("Dataset","OpenFF VEHICLe Set 1")
ds.get_values(program="psi4", keywords="default")
```

`gradient_dataset_fixture` should include keyword sets to catch this in the future.

Reply:
In general do we want to support non-aliased keywords through here? That is extra functionality which may be useful, but starts to become hard to tell what is an alias and what is an Id.

Reply:
This is still an issue. 

![image](https://user-images.githubusercontent.com/1508995/64979446-ba56a000-d885-11e9-8301-4f06d678b8e2.png)

Reply:
I think this solves the issue:

```
-            if isinstance(query_set["keywords"], str):
-                query_set["keywords"] = self.data.alias_keywords[query_set["program"]][query_set["keywords"]]
+            query_set["keywords"] = self.data.alias_keywords[query_set["program"]].get(query_set["keywords"], query_set["keywords"])
```

Reply:
Keywords must be an alias in general. I have patched this slightly, but I still don't understand the use case here.

Reply:
The problem was that this broke Dataset.get_values. It is now fixed.